File: algorithms-and-data-structures/algorithms-and-data-structures-2/data-structures/linear-data-structures/queue/priorityQueue.java
class Queue{
  int q,int n;
  int[] array;
  public Queue(int x){
    q=0;
    n=x;
    array=new int[x];
  }
  void enqueue(int x){
    if(q==n)return;
    int j=q-1;
    while(j>=0 && array[j]>x){
      array[j+1]=array[j];
      j--;
    }
    array[j+1]=x;
  }
}

File: algorithms-and-data-structures/algorithms-and-data-structures-2/data-structures/linear-data-structures/queue/priorityQueue.c
typedef struct{
  int q,n;
  int* array;
}Queue;
void set(Queue* queue,int n){
  (*queue).q=0;
  (*queue).n=x;
  (*queue).array=malloc(n*sizeof(int));
}
void enqueue(Queue* queue,int x){
  if((*queue).q==(*queue).n)return;
  int j=(*queue).q-1;
  while(j>=0 && (*queue).array[j]>x){
    (*queue).array[j+1]=(*queue).array[j];
    j--;
  }
  (queue).array[j+1]=x;
}
